src/assets/class-manifest.php

```diff
@@ -35,14 +99,56 @@ abstract class Manifest {
    *
    * @since 0.1.0
    */
-  public function register_assets_manifest_data() : string {
+  protected function get_raw_data() : string {
 
     $manifest = $this->get_manifest_url();
     if ( ! file_exists( $manifest ) ) {
-      $error_message = esc_html__( 'manifest.json is missing. Bundle the theme before using it.', 'developer-portal' );
+      $error_message = esc_html__( 'manifest.json is missing. Bundle the theme before using it.', 'eightshift-libs' );
       throw Exception\Missing_Manifest::message( $error_message );
     }
 
     return implode( ' ', file( $manifest ) );
   }
+
+  /**
+   * Provide manifest.json url location.
+   * If you are using a plugin or a different manifest location provide location with this method.
+   *
+   * @return string
+   *
+   * @since 0.6.0 Changed from abstract to predefined.
+   * @since 0.1.0
+   */
+  protected function get_manifest_url() : string {
+    return get_template_directory() . '/skin/public/manifest.json';
+  }
+
+  /**
+   * Return json_decoded manifest data, now you can call items by object key to get the value.
+   *
+   * @return object Manifest Object.
+   *
+   * @since 0.7.0 Changed to non static method.
+   * @since 0.6.0 Init
+   */
+  protected function get_decoded_manifest_data() {
+    $data = \json_decode( constant( $this->get_global_variable_name() ) );
+    if ( ! $data ) {
+      return null;
+    }
+
+    return $data;
+  }
+
+  /**
+   * Retrun string as a assets output prefix.
```

Comment:
```suggestion
   * Retrun string as an assets output prefix.
```

Can you also give an example of how this looks like in the end, in the comment? I'm not 100% sure what this does and why 🤷‍♂

Reply:
Manifest gives you and output like this:
```
{
  "application.css": "wp-content/themes/wp-boilerplate/skin/public/application-9d11e4277e8ca74d10ed.css",
  "application.js": "wp-content/themes/wp-boilerplate/skin/public/application.js",
  "applicationAdmin.css": "wp-content/themes/wp-boilerplate/skin/public/applicationAdmin-9d11e4277e8ca74d10ed.css",
  "applicationAdmin.js": "wp-content/themes/wp-boilerplate/skin/public/applicationAdmin.js",
  "jquery.min.js": "wp-content/themes/wp-boilerplate/skin/public/jquery.min.js",
  "logo.svg": "wp-content/themes/wp-boilerplate/skin/public/logo.svg"
}
```

this method adds site url to the final output url.
I have added home_url as a default (using it in theme), if you are using it in plugin you will override this method